homeassistant/components/roomba/irobot_base.py

```diff
@@ -219,6 +216,24 @@ def extra_state_attributes(self):
 
         return state_attrs
 
+    def get_cleaning_status(self, state) -> int:
+        if not (mission_state := state.get("cleanMissionStatus")):
+            return (0, 0)
+
+        if (cleaning_time := mission_state.get("mssnM", 0)) > 0:
+            pass
+        elif start_time := mission_state.get("mssnStrtTm"):
+            now = dt_util.as_timestamp(dt_util.utcnow())
+            if now > start_time:
+                cleaning_time = (now - start_time) // 60
+
+        if (cleaned_area := mission_state.get("sqft", 0)) > 0:  # Imperial
```

Comment:
```suggestion
        if cleaned_area := mission_state.get("sqft", 0):  # Imperial
```

Reply:
Ah, of course... I knew there was a reason I hadn't put those in initially 😅